src/journey-maps/angular/journey-maps.ts

```diff
@@ -501,9 +501,23 @@ export class SbbJourneyMaps implements OnInit, AfterViewInit, OnDestroy, OnChang
     }
 
     if (changes.poiOptions) {
-      this._executeWhenMapStyleLoaded(() =>
-        this._mapService.updatePoiVisibility(this._map, this.poiOptions)
-      );
+      this._executeWhenMapStyleLoaded(() => {
+        const poiEnvironmentChanged =
+          !changes.poiOptions.firstChange &&
+          changes.poiOptions.previousValue?.environment !==
+            changes.poiOptions.currentValue?.environment;
+
+        if (poiEnvironmentChanged) {
+          // Wait for Visibility to switch until new style got loaded
```

Comment:
In my opinion it's not necessary to reload the whole style. We also could replace the URL of the existing source.

    const source = this._map.getSource('journey-pois-source') as VectorTileSource;
    // PROD --> INT
    source.setUrl(source.url.replace('journey_pois', 'journey_pois_integration'));

Reply:
Implemented in newest commit.